server/src/main/scala/actors/PartitionedPortStateActor.scala

```diff
@@ -0,0 +1,210 @@
+package actors
+
+import actors.DrtStaticParameters.expireAfterMillis
+import actors.acking.AckingReceiver.{Ack, StreamCompleted, StreamFailure, StreamInitialized}
+import actors.daily.{TerminalDayQueuesUpdatesActor, TerminalDayStaffUpdatesActor, UpdatesSupervisor}
+import akka.actor.{Actor, ActorContext, ActorRef, ActorSystem, Props}
+import akka.pattern.{ask, pipe}
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import drt.shared.CrunchApi._
+import drt.shared.FlightsApi.{FlightsWithSplits, FlightsWithSplitsDiff}
+import drt.shared.Terminals.Terminal
+import drt.shared._
+import org.slf4j.{Logger, LoggerFactory}
+import services.crunch.deskrecs.{GetFlights, GetStateForDateRange, GetStateForTerminalDateRange, PortStateRequest}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.language.postfixOps
+
+object PartitionedPortStateActor {
+  def apply(now: () => SDateLike, airportConfig: AirportConfig, journalType: StreamingJournalLike, legacyDataCutoff: SDateLike)
+           (implicit system: ActorSystem, ec: ExecutionContext): ActorRef = {
+    val lookups: MinuteLookups = MinuteLookups(system, now, MilliTimes.oneDayMillis, airportConfig.queuesByTerminal)
```

Comment:
we should look at moving the system out of here and rather have MinuteLookups be an actor.